llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -6906,3 +6923,195 @@ bool CombinerHelper::matchOr(MachineInstr &MI, BuildFnTy &MatchInfo) {
 
   return false;
 }
+
+bool CombinerHelper::isZExtOrTruncLegal(LLT ToTy, LLT FromTy) const {
```

Comment:
If you're calling the variants `OrBeforeLegalizer` you should also make it clear in this function's name. Sometimes we want to know even before legalization whether something is actually legal.